agents/fsagent.go

```diff
@@ -298,7 +298,7 @@ func (fsa *FSsessions) Connect() error {
 		fsa.conns[connIdx] = fSock
 		utils.Logger.Info(fmt.Sprintf("<%s> successfully connected to FreeSWITCH at: <%s>", utils.FreeSWITCHAgent, connCfg.Address))
 		go func() { // Start reading in own goroutine, return on error
-			if err := fsa.conns[connIdx].ReadEvents(); err != nil {
+			if err := fSock.ReadEvents(); err != nil {
 				errChan <- err
 			}
 		}()
```

Comment:
Can you please add the fsock as a parameter to the goroutine like so:

```suggestion
		go func(fsock *fsock.FSock) { // Start reading in own goroutine, return on error
			if err := fsock.ReadEvents(); err != nil {
				errChan <- err
			}
		}(fSock)
```

Reply:
changed